dask_expr/_expr.py

```diff
@@ -3664,15 +3695,10 @@ def _check_dependents_are_predicates(
     return other_names.issubset(allowed_expressions)
 
 
-def calc_divisions_for_align(*exprs):
+def calc_divisions_for_align(*exprs, allow_shuffle=True):
```

Comment:
Do we use `allow_shuffle` anywhere?

Reply:
Yeah we still do in map_overlap, this is also on my bucket list after that one is in